Content.Server/Dragon/DragonSystem.cs

```diff
@@ -0,0 +1,195 @@
+
+            //NOTE: I honestly don't know much on how one detects valid eating targets, so right now I am using a body component to tell them apart
+            //That way dragons can't devour guardians and other dragons. Yet.
+
+            if (EntityManager.TryGetComponent(target, out MobStateComponent targetstate))
+            {
+                if (targetstate.CurrentState != null)
+                {
+                    //You can only devour dead or crit targets
+                    if (targetstate.CurrentState.IsIncapacitated())
+                    {
+                        if (EntityManager.TryGetComponent(dragonuid, out DamageableComponent dragonhealth))
+                        {
+                            //Humanoid devours allow dragon to get eggs, corpses included
+                            if (EntityManager.TryGetComponent(target, out HumanoidAppearanceComponent humanoid))
+                            {
+                                dragoncomp.EggsLeft++;
+                                //inject the healing chemical into the system. Yes the dragon just fucking drinks his own blood.
+                                _bloodstreamSystem.TryAddToChemicals(dragonuid, ichorInjection);
+                                //Sends the human entity into the stomach so it can be revived later. Withold further comments.
+                                dragoncomp.DragonStomach.Insert(target);
+                                SoundSystem.Play(Filter.Pvs(dragonuid), "/Audio/Effects/sound_magic_demon_consume.ogg");
+
+                            }
+                            //Non-humanoid mobs can only heal dragon for half the normal amount
+                            else
+                            {
+                                //heal HALF the damage
+                                _bloodstreamSystem.TryAddToChemicals(dragonuid, halfedIchorInjection);
+                                //Sends the non-human entity into the stomach
+                                //NOTE: I am a bit conflicted on this, and only really adding this because force-delete is bad, is this needed?
+                                dragoncomp.DragonStomach.Insert(target);
+                                SoundSystem.Play(Filter.Pvs(dragonuid), "/Audio/Effects/sound_magic_demon_consume.ogg");
+                            }
+                        }
+                        else return;
+
+                    }
+                    else _popupSystem.PopupEntity(Loc.GetString("devour-action-popup-message-fail-target-alive"), dragonuid, Filter.Entities(dragonuid));
+                    return;
+                }
+                else return;
+            }
+            // If it is a structure it goes through the motions of DoAfter
+            else if (EntityManager.TryGetComponent(target, out TagComponent tags))
+            {
+                // If it can be built- it can be destoryed
+                if (tags.Tags.Contains("RCDDeconstructWhitelist"))
```

Comment:
this is a weird way to tag it but why not. although it should really be using an `EntityWhitelist` on the component, rather than hardcoding the tag here, and doing `IsValid()` rather than `HasTag` or `Tags.Contains`. that way this can be tweaked in the future if we need to very easily